File: lab-programs/edgeDetection.m
%% Edge detection

I=imread('D:\NSTU\DIP-materials\lab-programs\image\saturn.png');
I=rgb2gray(I);

sEdge=edge(I,'sobel');
prEdge=edge(I,'prewitt');
roEdge=edge(I,'roberts');

subplot(2,2,1);
imshow(I);
title('Original Image');

subplot(2,2,2);
imshow(sEdge);
title('Sobel');

subplot(2,2,3);
imshow(prEdge);
title('Prewitt');

subplot(2,2,4);
imshow(roEdge);
title('Roberts');

File: Practice/opencv_lab/edgedetection.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

# Read the image
I = cv2.imread('test.jpg')
I = cv2.cvtColor(I, cv2.COLOR_BGR2GRAY)

# Edge detection using different filters
sEdge = cv2.Sobel(I, cv2.CV_64F, 1, 1, ksize=3)
prEdge = cv2.Prewitt(I, cv2.CV_64F)
roEdge = cv2.Roberts(I, cv2.CV_64F)

# Plotting the results
plt.figure(figsize=(10, 8))

plt.subplot(2, 2, 1)
plt.imshow(I, cmap='gray')
plt.title('Original Image')

plt.subplot(2, 2, 2)
plt.imshow(np.abs(sEdge), cmap='gray')
plt.title('Sobel')

plt.subplot(2, 2, 3)
plt.imshow(np.abs(prEdge), cmap='gray')
plt.title('Prewitt')

plt.subplot(2, 2, 4)
plt.imshow(np.abs(roEdge), cmap='gray')
plt.title('Roberts')

plt.show()
